### PR TITLE
Version Check 

### PR DESCRIPTION
**Pull request by cgon50**
- Created a serverinfo class that will hold the version of the server being ran
- updated chat to enable 'info' as a command that returns the version of the server 
- also updated the help command to include info

### CODE REVIEW

**Review by cgon50 — COMMENTED** [review]
Please let me know if you guys want more comments or any extra explanation on any of the changes

File: src/codeu/chat/client/commandline/Chat.java
Comment (tsums) [inline src/codeu/chat/client/commandline/Chat.java]: Leave a space between the `//` and your comment text.

File: src/codeu/chat/client/core/Context.java
Comment (tsums) [inline src/codeu/chat/client/core/Context.java]: Prefer to write comments that are well-formatted English sentences, with a capitalized first letter and period at the end.

I don't want to spam the review by commenting on each one, but this applies elsewhere as well.

File: src/codeu/chat/client/core/View.java
Comment (tsums) [inline src/codeu/chat/client/core/View.java]: This comment isn't very useful - I can infer what it tells me from the method signature. Consider providing an overview of the method - how it does what it does, what possible exceptions it could encounter, etc.

If you want to go the extra mile, use [Javadoc](https://en.wikipedia.org/wiki/Javadoc).

File: src/codeu/chat/client/core/View.java
Comment (tsums) [inline src/codeu/chat/client/core/View.java]: Is there code missing here? Does this comment refer to the line of code above it? Generally if you're describing a line of code, place the comment above the line being described.

File: src/codeu/chat/client/core/View.java
Comment (tsums) [inline src/codeu/chat/client/core/View.java]: It might be useful to attempt to `getMessage()` from the exception and print it. The error is unknown to us now, but at runtime we can give the user some hint as to what happened.

File: src/codeu/chat/common/BasicView.java
Comment (tsums) [inline src/codeu/chat/common/BasicView.java]: Extra whitespace?

File: src/codeu/chat/common/ServerInfo.java
Comment (tsums) [inline src/codeu/chat/common/ServerInfo.java]: Incomplete comment.

File: src/codeu/chat/server/Server.java
Comment (tsums) [inline src/codeu/chat/server/Server.java]: Similar to before, this comment isn't very useful. Consider adding detail, javadoc, or remove altogether.

File: src/codeu/chat/server/View.java
Comment (tsums) [inline src/codeu/chat/server/View.java]: Why does this return null? Seems like a good place for a comment, or maybe an `UnsupportedOperationException` if this isn't supposed to be called ever.

File: src/codeu/chat/common/ServerInfo.java
Comment (tsums) [inline src/codeu/chat/common/ServerInfo.java]: Looks like you're parsing `SERVER_VERSION`, not a user id.

File: src/codeu/chat/common/BasicView.java
Comment (tsums) [inline src/codeu/chat/common/BasicView.java]: Document the purpose of this method. In interfaces, especially since there is no method body to speak for itself, it's important to understand the motivation / usage of a method, especially since consumers of this interface may not have visibility into the implementation code.

**Review by tsums — DISMISSED** [review]
Consider using a tool like [google-java-format](https://github.com/google/google-java-format) to apply consistent formatting to your code for you. Should make it easier to read.

File: src/codeu/chat/common/ServerInfo.java
Comment (cgon50) [reply src/codeu/chat/common/ServerInfo.java]: The tutorial for the technical activities provided that code, I just added the try catch 
image

File: src/codeu/chat/client/core/View.java
Comment (cgon50) [reply src/codeu/chat/client/core/View.java]: Fixed

File: src/codeu/chat/client/core/View.java
Comment (cgon50) [reply src/codeu/chat/client/core/View.java]: fixed

File: src/codeu/chat/client/commandline/Chat.java
Comment (cgon50) [reply src/codeu/chat/client/commandline/Chat.java]: fixed

File: src/codeu/chat/client/core/Context.java
Comment (cgon50) [reply src/codeu/chat/client/core/Context.java]: Fixed

File: src/codeu/chat/common/BasicView.java
Comment (cgon50) [reply src/codeu/chat/common/BasicView.java]: fixed

File: src/codeu/chat/common/BasicView.java
Comment (cgon50) [reply src/codeu/chat/common/BasicView.java]: fixed

File: src/codeu/chat/server/Server.java
Comment (cgon50) [reply src/codeu/chat/server/Server.java]: fixed

File: src/codeu/chat/common/ServerInfo.java
Comment (tsums) [reply src/codeu/chat/common/ServerInfo.java]: My apologies, I should have clarified. I'm commenting specifically on the string you're printing in line 17, `"Server could not parse the user id"`. Are we parsing a user id?

Comment (cgon50) [reply src/codeu/chat/common/ServerInfo.java]: fixed, thanks for the clarification!

File: src/codeu/chat/server/View.java
Comment (cgon50) [reply src/codeu/chat/server/View.java]: fixed

File: src/codeu/chat/common/ServerInfo.java
Comment (cgon50) [reply src/codeu/chat/common/ServerInfo.java]: fixed

File: build.py
Comment (tsums) [inline build.py]: What's the context for these changes to the build script?

Comment (cgon50) [reply build.py]: Just added parenthesis since I thought python print statements needed them

Comment (tsums) [reply build.py]: This script is designed for Python 2.x, which does not require parentheses. If you look, there's other instances in the build file which do not use parentheses. 

Since the change is unrelated to this pull request, let's revert it to keep things lean.

Comment (cgon50) [reply build.py]: Reverted

**Comment by tsums** [comment]
LGTM. Wait for @rcarla25 to review before merging.

File: src/codeu/chat/client/core/View.java
Comment (cprivera) [inline src/codeu/chat/client/core/View.java:141]: I'm confused on what is going on in this method, specifically why the `'try'` block starts with `(final Connection connection = this.source.connect())`

**Pull request merged**
